### PR TITLE
Feature/moveit object attachment

### PR DESCRIPTION
**Pull request by juandhv**
This includes the changes for disabling/enabling objects in the vicon_object_server

### CODE REVIEW

File: vicon_object_server_library/src/scene.cpp [hunk 1]
```diff
@@ -147,15 +147,10 @@ Object *Scene::getObject(const std::string &name)
...
-{
-    return getObject(name);
-}
-
 void Scene::removeObject(const std::string &name)
 {
     auto found = findGetName<Object>(objects_, name);
-    if (found != objects_.end())
+    if (found == objects_.end())
         throw Exception(1, "Scene::removeObject: Object not found!");
```
Comment (BryceStevenWilley) [inline vicon_object_server_library/src/scene.cpp]: I know this isn't the point of this PR, but this call still makes the server crash, correct?

File: vicon_object_server_objects/objects/table_kavrakilab.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
-name            : "table_kavrakilab"
+name            : "table1"
 frame           : "vicon/Table/Table"
 geometry        :
   type          : "box"
-  dimensions    : [0.75 , 1.5, 0.15]
+  dimensions    : [0.75 , 1.5, 0.13]
```
Comment (BryceStevenWilley) [inline vicon_object_server_objects/objects/table_kavrakilab.yaml]: Any reason you make the table model smaller? I'd imagine that it'd be better to keep extra padding around the table.

File: vicon_object_server_library/src/scene.cpp (hunk 1 above)
Comment (BryceStevenWilley) [reply vicon_object_server_library/src/scene.cpp]: Okay, I'm gonna make an issue for that.

File: vicon_object_server_server/src/main.cpp [hunk 3]
```diff
@@ -72,39 +72,52 @@ void MoveItInteraction::callback(const moveit_msgs::PlanningScene& msg)
...
-void MoveItInteraction::update(const Scene& scene)
+void MoveItInteraction::update(Scene& scene)
 {
     std::unique_lock<std::mutex> lock(mutex_);
 
-    scene_ = moveit_msgs::PlanningScene();
+    planning_scene_ = moveit_msgs::PlanningScene();
 
-    for (auto obj : scene.getObjects())
+    //std::cout << "There are " << scene.getObjectNames().size() << " names" << std::endl;
```
Comment (zkingston) [inline vicon_object_server_server/src/main.cpp]: Could we remove these debugging statements?

File: vicon_object_server_server/src/main.cpp [hunk 4]
```diff
@@ -72,39 +72,52 @@ void MoveItInteraction::callback(const moveit_msgs::PlanningScene& msg)
...
-        auto attached = std::find(attached_objects_.begin(), attached_objects_.end(), obj.getName()) != attached_objects_.end();
-        auto frame = getObjectFrame(&obj);
-        bool exists = obj_in_scene != objects_.end();
-
-        // If not within the scene, should add the geometry.
-        if (!exists)
-            scene_.world.collision_objects.push_back(obj.getCollisionObject(frame));
-        else
-            scene_.world.collision_objects.push_back(obj.getCollisionObject(frame, true));
+    	//std::cout << "MOVEIT" << std::endl;
```
Comment (zkingston) [inline vicon_object_server_server/src/main.cpp]: See above

File: vicon_object_server_server/src/main.cpp [hunk 5]
```diff
@@ -72,39 +72,52 @@ void MoveItInteraction::callback(const moveit_msgs::PlanningScene& msg)
...
+    	//std::cout << "object status: " << obj->getStatus() << std::endl;
+
+    	bool exists = std::find_if(objects_.begin(), objects_.end(),
+    			[&](std::string& x) { return obj_name.compare(x) == 0; }) != objects_.end();
+    	bool attached = std::find(attached_objects_.begin(), attached_objects_.end(), obj_name) != attached_objects_.end();
+    	const std::string& frame = getObjectFrame(obj);
+
+    	// If not within the scene, should add the geometry.
+    	if (!exists)
+    		planning_scene_.world.collision_objects.push_back(obj->getCollisionObject(frame));
```
Comment (zkingston) [inline vicon_object_server_server/src/main.cpp]: It looks like the indentation was changed here? Can we make this consistent and run clang-format?

File: vicon_object_server_server/src/main.cpp [hunk 6]
```diff
@@ -160,31 +173,37 @@ void VICONObjectServer::update()
         auto obj_frame = getObjectFrame(obj);
         try
         {
-            auto object_pose =
-                tf2::transformToEigen(lookup(base_frame_, obj->getVICONFrame()));
+        	auto object_pose =
```
Comment (zkingston) [inline vicon_object_server_server/src/main.cpp]: Same here, strange indentation changes.

File: vicon_object_server_server/src/main.cpp [hunk 7]
```diff
@@ -160,31 +173,37 @@ void VICONObjectServer::update()
...
-            tf.header.frame_id = scene_.getURDFFrame();
-            tf.child_frame_id = obj_frame;
+        	// Offset is from the URDF base to the vicon base.
+        	tf_broadcast_.sendTransform(tf);
+        	obj->setActive(true);
 
-            // Offset is from the URDF base to the vicon base.
-            tf_broadcast_.sendTransform(tf);
-            obj->setActive(true);
+        	//std::cout << "TF" << std::endl;
```
Comment (zkingston) [inline vicon_object_server_server/src/main.cpp]: Remove debugging

File: vicon_object_server_server/src/main.cpp (hunk 3 above)
Comment (juandhv) [reply vicon_object_server_server/src/main.cpp]: I thought I did. Done

File: vicon_object_server_server/src/main.cpp (hunk 4 above)
Comment (juandhv) [reply vicon_object_server_server/src/main.cpp]: I thought I did. Done

File: vicon_object_server_server/src/main.cpp (hunk 5 above)
Comment (juandhv) [reply vicon_object_server_server/src/main.cpp]: My bad. I was using eclipse built-in format. I have setup eclipse with clang-format.

File: vicon_object_server_server/src/main.cpp (hunk 6 above)
Comment (juandhv) [reply vicon_object_server_server/src/main.cpp]: Done

File: vicon_object_server_server/src/main.cpp (hunk 7 above)
Comment (juandhv) [reply vicon_object_server_server/src/main.cpp]: Done

File: vicon_object_server_library/include/vicon_object_client.h [hunk 8]
```diff
@@ -98,6 +100,16 @@ namespace vicon_object_server
         // Throws Exception if service call fails
         int objectCollision(const std::string& name, bool collision);
 
+        // Returns 0 upon success,
```
[the remark this answers is not in the document]
Comment (juandhv) [reply vicon_object_server_library/include/vicon_object_client.h]: Ups .. I didn't see it before. I was looking for Disable; I could never have imagined objectCollision meant that :-). Should I remove objectCollision now? or we can leave it for a next and cleaner version ?

[the remark this answers is not in the document]
Comment (BryceStevenWilley) [reply vicon_object_server_library/include/vicon_object_client.h]: Might as well remove it now, I don't think anyone's using it at the moment.

[the remark this answers is not in the document]
Comment (juandhv) [reply vicon_object_server_library/include/vicon_object_client.h]: Done

**Pull request merged**